### ISSUE TITLE
template filter click with js error

### ISSUE THREAD

**Issue by archonwang** [akiko-pusu/redmine_issue_templates#33]
New issue with template, when I click template filter, there's nothing happened. 
check the console with error s follows: 
> 
> VM6188:23 GET http://192.168.1.41/projects/demo/issue_templates/list_templates?issue_tracker_id=1 500 (Internal Server Error)
> send 
> @ jquery-1.11.1-ui-1.11.0-ujs-3.1.4.js:4ajax 
> @ jquery-1.11.1-ui-1.11.0-ujs-3.1.4.js:4openDialog 
> @ issue_templates.js:29(anonymous function) 
> @ VM6188:23dispatch 
> @ jquery-1.11.1-ui-1.11.0-ujs-3.1.4.js:3r.handle 
> @ jquery-1.11.1-ui-1.11.0-ujs-3.1.4.js:3
> VM6188:23 XHR finished loading: GET "http://192.168.1.41/projects/demo/issue_templates/list_templates?issue_tracker_id=1".
>send 
> @ jquery-1.11.1-ui-1.11.0-ujs-3.1.4.js:4ajax
> @ jquery-1.11.1-ui-1.11.0-ujs-3.1.4.js:4openDialog 
> @ issue_templates.js:29(anonymous function)
> @ VM6188:23dispatch 
> @ jquery-1.11.1-ui-1.11.0-ujs-3.1.4.js:3r.handle 
> @ jquery-1.11.1-ui-1.11.0-ujs-3.1.4.js:3
> 

the log file with error
```
F, [2016-06-22T13:28:47.948590 #32444] FATAL -- : 
ActionView::Template::Error (undefined method `each' for nil:NilClass):
    26:         </td>
    27:       </tr>
    28: <% end %>
    29: <% inherit_templates.each do |template| %>
    30:       <tr class="<%= cycle('odd', 'even') %> template_data">
    31:         <td>
    32:           <%= template.title %>
  plugins/redmine_issue_templates/app/views/issue_templates/_list_templates.html.erb:29:in `_plugins_redmine_issue_templates_app_views_issue_templates__list_templates_html_erb___4221372361586885076_70188302535700'
  plugins/redmine_issue_templates/app/controllers/issue_templates_controller.rb:170:in `block (2 levels) in list_templates'
  plugins/redmine_issue_templates/app/controllers/issue_templates_controller.rb:168:in `list_templates'
  lib/redmine/sudo_mode.rb:63:in `sudo_mode'
```

I use redmine 3.2 and chrome. Please check it. Thanks.

**Comment by akiko-pusu** [comment]
In my env, this issue does not happen.
So, fitst, please open http://192.168.1.41/projects/demo/issue_templates/list_templates?issue_tracker_id=1 directory with your browser.
If controller works fine, simple table html is returned, like this.

```
<table class="list template_list">
  <tr>
    <th>Issue title</th>
    <th>Issue body</th>
    <th>Default value</th>
    <th>Apply</th>
  </tr>
      <tr class="odd template_data">
        <td>
          feature template
        </td>
        <td>
          <div class="template_tooltip_wrapper">
            <a class="icon template_tooltip" href="#" title="Preview"></a>
            <div class="wiki template_tooltip_body">
              <span class="title">feature template</span>
              <p>feature template bidy is here!!.......</p>
            </div>
          </div>
        </td>
        <td class="center"><span class="icon-only icon-checked"></span></td>
        <td>
          <a href="#" onclick="updateSelect(1, false); return false;"
             class="icon icon-test"></a>
        </td>
      </tr>
      <tr class="even template_data">
        <td>
          Test Global Template for Feature
        </td>
        <td>
          <div class="template_tooltip_wrapper">
            <a class="icon template_tooltip" href="#" title="Preview"></a>
            <div class="wiki template_tooltip_body">
              <span class="title">Test Global Template for Feature</span>
              <p>This is a template to test Global Template.</p>
            </div>
          </div>
        </td>
        <td class="center"></td>
        <td>
          <a href="#" onclick="updateSelect(4, true); return false;"
             class="icon icon-test"></a>
        </td>
      </tr>
</table>
```

If the situation is the same, try to remove line 29 - 49, inherit_templates.each do  ... end block of views/issue_templates/_list_templates.html.erb.

**Comment by archonwang** [comment]
@akiko-pusu , I will test it later. Thanks.

the full trace error log is 


> An ActionView::Template::Error occurred in issue_templates#list_templates:
>  
>   undefined method `each' for nil:NilClass
>   plugins/redmine_issue_templates/app/views/issue_templates/_list_templates.html.erb:29:in `_plugins_redmine_issue_templates_app_views_issue_templates__list_templates_html_erb__182501847570646770_70340047864360'
>  
>  
> -------------------------------
> Request:
> -------------------------------
>  
> * URL : http://redmine.utranshub.com/projects/uth_testteam/issue_templates/list_templates?issue_tracker_id=5
>   * HTTP Method: GET
>   * IP address : 192.168.60.239
>   * Parameters : {"issue_tracker_id"=>"5", "controller"=>"issue_templates", "project_id"=>"uth_testteam", "action"=>"list_templates"}
>   * Timestamp  : 2016-06-22 14:44:48 UTC
>   * Server : jira
>   * Rails root : /opt/redmine-3.3
>   * Process: 5938
>  
> -------------------------------
> Session:
> -------------------------------
>  
>   * session id: "88fc20ab8adb69594abfcb7b809af43e"
>   * data: {"session_id"=>"88fc20ab8adb69594abfcb7b809af43e",
>    "user_id"=>"8",
>    "tk"=>"0cf491fe95ea9bc74763f2fcc3df82f2126e630d",
>    "sudo_timestamp"=>1466604255,
>    "my_page_sort"=>"none",
>    "_csrf_token"=>"bOReldjXnfac+KYCwDnpQEhBCfRkON7jXI49ti8jAdM=",
>    "recent_projects"=>[2, 1, 15, 11, 7],
>    "admin_info_show_sidebar"=>true,
>    "spectator_id"=>1,
>    "my_account_sort"=>"none",
>    "query"=>
>     {:project_id=>15,
>      :filters=>{"status_id"=>{:operator=>"o", :values=>[""]}},
>      :group_by=>"",
>      :column_names=>nil,
>      :totalable_names=>[:estimated_hours, :spent_hours]},
>    "issues_index_sort"=>"id:desc",
>    "meetings_index_sort"=>"",
>    "evms_index_show_sidebar"=>true}
>  
> -------------------------------
> Environment:
> -------------------------------
>  
>   * GATEWAY_INTERFACE                              : CGI/1.2
>   * HTTP_ACCEPT                                    : */*
>   * HTTP_ACCEPT_ENCODING                           : gzip, deflate, sdch
>   * HTTP_ACCEPT_LANGUAGE                           : zh-CN,zh;q=0.8
>   * HTTP_CONNECTION                                : keep-alive
>   * HTTP_COOKIE                                    : _redmine_session=aUhUVStSWXA1Tmo5TURvd2dPcU16aXhhdEVHampoUXh5S2J3eWk1Z1hmUjlkU1crSWJZb2ZUbU1NaWJpS2VxTVJpNTc5c3ZiMWhhUTBDTlRwYmZPODJueTJCTy9IODVYdDloTUdPdTErNWNtcDJpbC9Yd3Y5c3c5dXpuMnY3dDhObVhyWjRocjdudURjMmwvWEtQMk5KeTEzMW5nVitORnlRakxqVXIxRVhwV3Rkay95ZVBGeGgvV2xrdHJUU3h0ZG9OWU40b3RSd0UzcDdBeTgxRjA0MWRaTjVlNnVGWm8rOXpmbWZOcUFOR1B2a0RJMHB1TmU1TXR2RnhuRjNtU1hzZWw2UU9mOGVSbU5PS0tPVnllOWZzZjBlbEQyWnB3eWhpU3hFSHhWZnAzbFFKYjFSY3cwOHhWbWhaVGt3Z1BBMjNJR3pVTXNNUEc1clpZZTNSQzFRc0ozSmk4MElqNktsdUpOUklFVkdkMHNjS2RMdEdBR3k4SmkwOElndWRuc211bWhMVWlrQUJBelB6Y1dVbFBCYmF4dG14aW1obE44bEgxYlNBNkZSN1BHUUM1WThQeE8vRGF6YWI0a05MZG5jeHRtL0RrWHJTY1NLRUtWY3d1Q3lBT1R5dHRnS2Y2MEZiampLb0w5SlphZVJsTk9nUXBZRURIZDVPdXp3MXZKZ0ZVWXEwZlNhSTB2bEVoY3FqUnRaWGdGaWZUWXdtRXJVR1htOHh3UU5rQ09EN3hHMDdxTEhxUWJrdW5MUncwN0w2KzFHbktTTGRQMUJjdmVuMytqdGhIdlEyRDhGU25Ed0RtbHdTU0dYR2dMYUs3V3JtanFBMDBseGtqMDMzaFQvVjc2T3lnOHZHZzZHOTRGU0NpK21IVnd6cDNjcnYxajBxc3hpemExQUljVy90WGxJVEUyU1BxR3E0a3JTRnQrYkgzUk5BcWhRb0VQMXBMdkxuci92VWJtQm56ejI3WnFCdlp5eGJQWkozcVQ3ckhlNDZrQkgxa2xvNE5BUzVwSW1XZVZaeWx4Mko1a3JZVGw2SFFvNStMTnNpakJ5M204bGxEQVVqRFZlTldyUSsvZzVXQ3VKRUkxVmJ0SU9nL3JnMExIczJNRTJCdTNrQmlEOFdHdktzdE1sY0hvaXBmTlNiU3RHaC9zcE9YWHlHclVmb3BVb1NadTVVOHY5d08tLXIydkNHOUNQQWdyUmh4N2pGZ05YMWc9PQ%3D%3D--600404b2fee2ea0df8e56e38c46507acb41410f0
>   * HTTP_HOST                                      : redmine.utranshub.com
> * HTTP_REFERER : http://redmine.utranshub.com/projects/uth_testteam/issues/new
>   * HTTP_USER_AGENT                                : Mozilla/5.0 (Windows NT 6.1; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/51.0.2704.103 Safari/537.36
>   * HTTP_VERSION                                   : HTTP/1.1
>   * HTTP_X_CSRF_TOKEN                              : vs8dBxvrrkxRKz4V+mwdJr733DX8dxpwdPuu8OMWmAbSK0OSwzwzus3TmBc6VfRm9rbVwZhPxJModZNGzDWZ1Q==
>   * HTTP_X_REQUESTED_WITH                          : XMLHttpRequest
>   * ORIGINAL_FULLPATH                              : /projects/uth_testteam/issue_templates/list_templates?issue_tracker_id=5
>   * ORIGINAL_SCRIPT_NAME                           :
>   * PATH_INFO                                      : /projects/uth_testteam/issue_templates/list_templates
>   * QUERY_STRING                                   : issue_tracker_id=5
>   * REMOTE_ADDR                                    : 192.168.60.239
>   * REQUEST_METHOD                                 : GET
>   * REQUEST_PATH                                   : /projects/uth_testteam/issue_templates/list_templates
>   * REQUEST_URI                                    : /projects/uth_testteam/issue_templates/list_templates?issue_tracker_id=5
>   * ROUTES_51839660_SCRIPT_NAME                    :
>   * SCRIPT_NAME                                    :
>   * SERVER_NAME                                    : redmine.utranshub.com
>   * SERVER_PORT                                    : 80
>   * SERVER_PROTOCOL                                : HTTP/1.1
>   * SERVER_SOFTWARE                                : thin 1.7.0 codename Dunder Mifflin
>   * action_controller.instance                     : #<IssueTemplatesController:0x007ff2702c0870>
>   * action_dispatch.backtrace_cleaner              : #<Rails::BacktraceCleaner:0x0000000534b200>
>   * action_dispatch.cookies                        : #<ActionDispatch::Cookies::CookieJar:0x007ff2a322e370>
>   * action_dispatch.cookies_digest                 :
>   * action_dispatch.cookies_serializer             :
>   * action_dispatch.encrypted_cookie_salt          : encrypted cookie
>   * action_dispatch.encrypted_signed_cookie_salt   : signed encrypted cookie
>   * action_dispatch.http_auth_salt                 : http authentication
>   * action_dispatch.key_generator                  : #<ActiveSupport::CachingKeyGenerator:0x00000005da1c68>
>   * action_dispatch.logger                         : #<ActiveSupport::Logger:0x0000000615db68>
>   * action_dispatch.parameter_filter               : [:password]
>   * action_dispatch.redirect_filter                : []
>   * action_dispatch.remote_ip                      : 192.168.60.239
>   * action_dispatch.request.accepts                : [#<Mime::Type:0x007ff2a322f568 @synonyms=[], @symbol=nil, @string="*/*", @hash=-3072026020700112302>]
>   * action_dispatch.request.content_type           :
>   * action_dispatch.request.formats                : [#<Mime::Type:0x007ff2a322f568 @synonyms=[], @symbol=nil, @string="*/*", @hash=-3072026020700112302>]
>   * action_dispatch.request.parameters             : {"issue_tracker_id"=>"5", "controller"=>"issue_templates", "project_id"=>"uth_testteam", "action"=>"list_templates"}
>   * action_dispatch.request.path_parameters        : {:controller=>"issue_templates", :project_id=>"uth_testteam", :action=>"list_templates"}
>   * action_dispatch.request.query_parameters       : {"issue_tracker_id"=>"5"}
>   * action_dispatch.request.request_parameters     : {}
>   * action_dispatch.request.unsigned_session_cookie: {"session_id"=>"88fc20ab8adb69594abfcb7b809af43e", "user_id"=>"8", "tk"=>"0cf491fe95ea9bc74763f2fcc3df82f2126e630d", "sudo_timestamp"=>1466604255, "my_page_sort"=>"none", "_csrf_token"=>"bOReldjXnfac+KYCwDnpQEhBCfRkON7jXI49ti8jAdM=", "recent_projects"=>[2, 1, 15, 11, 7], "admin_info_show_sidebar"=>true, "spectator_id"=>1, "my_account_sort"=>"none", "query"=>{:project_id=>15, :filters=>{"status_id"=>{:operator=>"o", :values=>[""]}}, :group_by=>"", :column_names=>nil, :totalable_names=>[:estimated_hours, :spent_hours]}, "issues_index_sort"=>"id:desc", "meetings_index_sort"=>"", "evms_index_show_sidebar"=>true}
>   * action_dispatch.request_id                     : 1d355475-dcf9-467d-b258-6c4101873a9f
>   * action_dispatch.routes                         : #<ActionDispatch::Routing::RouteSet:0x000000062e0558>
>   * action_dispatch.secret_key_base                : 59dda17a42803617f96151518b9aeecb94e132c6f140123f1982e1c8ae15d0103ffa854c4d8b0019
>   * action_dispatch.secret_token                   :
>   * action_dispatch.show_detailed_exceptions       : false
>   * action_dispatch.show_exceptions                : true
>   * action_dispatch.signed_cookie_salt             : signed cookie
>   * async.callback                                 : #<Method: Thin::Connection#post_process>
>   * async.close                                    : #<EventMachine::DefaultDeferrable:0x007ff2a3234d10>
>   * rack.errors                                    : #<IO:0x00000001ad5588>
>   * rack.input                                     : #<StringIO:0x007ff2a32358a0>
>   * rack.multiprocess                              : false
>   * rack.multithread                               : true
>   * rack.request.cookie_hash                       : {"_redmine_session"=>"aUhUVStSWXA1Tmo5TURvd2dPcU16aXhhdEVHampoUXh5S2J3eWk1Z1hmUjlkU1crSWJZb2ZUbU1NaWJpS2VxTVJpNTc5c3ZiMWhhUTBDTlRwYmZPODJueTJCTy9IODVYdDloTUdPdTErNWNtcDJpbC9Yd3Y5c3c5dXpuMnY3dDhObVhyWjRocjdudURjMmwvWEtQMk5KeTEzMW5nVitORnlRakxqVXIxRVhwV3Rkay95ZVBGeGgvV2xrdHJUU3h0ZG9OWU40b3RSd0UzcDdBeTgxRjA0MWRaTjVlNnVGWm8rOXpmbWZOcUFOR1B2a0RJMHB1TmU1TXR2RnhuRjNtU1hzZWw2UU9mOGVSbU5PS0tPVnllOWZzZjBlbEQyWnB3eWhpU3hFSHhWZnAzbFFKYjFSY3cwOHhWbWhaVGt3Z1BBMjNJR3pVTXNNUEc1clpZZTNSQzFRc0ozSmk4MElqNktsdUpOUklFVkdkMHNjS2RMdEdBR3k4SmkwOElndWRuc211bWhMVWlrQUJBelB6Y1dVbFBCYmF4dG14aW1obE44bEgxYlNBNkZSN1BHUUM1WThQeE8vRGF6YWI0a05MZG5jeHRtL0RrWHJTY1NLRUtWY3d1Q3lBT1R5dHRnS2Y2MEZiampLb0w5SlphZVJsTk9nUXBZRURIZDVPdXp3MXZKZ0ZVWXEwZlNhSTB2bEVoY3FqUnRaWGdGaWZUWXdtRXJVR1htOHh3UU5rQ09EN3hHMDdxTEhxUWJrdW5MUncwN0w2KzFHbktTTGRQMUJjdmVuMytqdGhIdlEyRDhGU25Ed0RtbHdTU0dYR2dMYUs3V3JtanFBMDBseGtqMDMzaFQvVjc2T3lnOHZHZzZHOTRGU0NpK21IVnd6cDNjcnYxajBxc3hpemExQUljVy90WGxJVEUyU1BxR3E0a3JTRnQrYkgzUk5BcWhRb0VQMXBMdkxuci92VWJtQm56ejI3WnFCdlp5eGJQWkozcVQ3ckhlNDZrQkgxa2xvNE5BUzVwSW1XZVZaeWx4Mko1a3JZVGw2SFFvNStMTnNpakJ5M204bGxEQVVqRFZlTldyUSsvZzVXQ3VKRUkxVmJ0SU9nL3JnMExIczJNRTJCdTNrQmlEOFdHdktzdE1sY0hvaXBmTlNiU3RHaC9zcE9YWHlHclVmb3BVb1NadTVVOHY5d08tLXIydkNHOUNQQWdyUmh4N2pGZ05YMWc9PQ==--600404b2fee2ea0df8e56e38c46507acb41410f0"}
>   * rack.request.cookie_string                     : _redmine_session=aUhUVStSWXA1Tmo5TURvd2dPcU16aXhhdEVHampoUXh5S2J3eWk1Z1hmUjlkU1crSWJZb2ZUbU1NaWJpS2VxTVJpNTc5c3ZiMWhhUTBDTlRwYmZPODJueTJCTy9IODVYdDloTUdPdTErNWNtcDJpbC9Yd3Y5c3c5dXpuMnY3dDhObVhyWjRocjdudURjMmwvWEtQMk5KeTEzMW5nVitORnlRakxqVXIxRVhwV3Rkay95ZVBGeGgvV2xrdHJUU3h0ZG9OWU40b3RSd0UzcDdBeTgxRjA0MWRaTjVlNnVGWm8rOXpmbWZOcUFOR1B2a0RJMHB1TmU1TXR2RnhuRjNtU1hzZWw2UU9mOGVSbU5PS0tPVnllOWZzZjBlbEQyWnB3eWhpU3hFSHhWZnAzbFFKYjFSY3cwOHhWbWhaVGt3Z1BBMjNJR3pVTXNNUEc1clpZZTNSQzFRc0ozSmk4MElqNktsdUpOUklFVkdkMHNjS2RMdEdBR3k4SmkwOElndWRuc211bWhMVWlrQUJBelB6Y1dVbFBCYmF4dG14aW1obE44bEgxYlNBNkZSN1BHUUM1WThQeE8vRGF6YWI0a05MZG5jeHRtL0RrWHJTY1NLRUtWY3d1Q3lBT1R5dHRnS2Y2MEZiampLb0w5SlphZVJsTk9nUXBZRURIZDVPdXp3MXZKZ0ZVWXEwZlNhSTB2bEVoY3FqUnRaWGdGaWZUWXdtRXJVR1htOHh3UU5rQ09EN3hHMDdxTEhxUWJrdW5MUncwN0w2KzFHbktTTGRQMUJjdmVuMytqdGhIdlEyRDhGU25Ed0RtbHdTU0dYR2dMYUs3V3JtanFBMDBseGtqMDMzaFQvVjc2T3lnOHZHZzZHOTRGU0NpK21IVnd6cDNjcnYxajBxc3hpemExQUljVy90WGxJVEUyU1BxR3E0a3JTRnQrYkgzUk5BcWhRb0VQMXBMdkxuci92VWJtQm56ejI3WnFCdlp5eGJQWkozcVQ3ckhlNDZrQkgxa2xvNE5BUzVwSW1XZVZaeWx4Mko1a3JZVGw2SFFvNStMTnNpakJ5M204bGxEQVVqRFZlTldyUSsvZzVXQ3VKRUkxVmJ0SU9nL3JnMExIczJNRTJCdTNrQmlEOFdHdktzdE1sY0hvaXBmTlNiU3RHaC9zcE9YWHlHclVmb3BVb1NadTVVOHY5d08tLXIydkNHOUNQQWdyUmh4N2pGZ05YMWc9PQ%3D%3D--600404b2fee2ea0df8e56e38c46507acb41410f0
>   * rack.request.query_hash                        : {"issue_tracker_id"=>"5"}
>   * rack.request.query_string                      : issue_tracker_id=5
>   * rack.run_once                                  : false
>   * rack.session                                   : #<ActionDispatch::Request::Session:0x007ff2702c29b8>
>   * rack.session.options                           : #<ActionDispatch::Request::Session::Options:0x007ff2702c2968>
>   * rack.url_scheme                                : http
>   * rack.version                                   : [1, 0]
>  
> -------------------------------
> Backtrace:
> -------------------------------
>  
>   plugins/redmine_issue_templates/app/views/issue_templates/_list_templates.html.erb:29:in `_plugins_redmine_issue_templates_app_views_issue_templates__list_templates_html_erb__182501847570646770_70340047864360'
>   plugins/redmine_issue_templates/app/controllers/issue_templates_controller.rb:170:in `block (2 levels) in list_templates'
>   plugins/redmine_issue_templates/app/controllers/issue_templates_controller.rb:168:in `list_templates'
>   lib/redmine/sudo_mode.rb:63:in `sudo_mode'
>

**Comment by archonwang** [comment]
I used the new commit and test it on Chrome, it worked well, Thanks @akiko-pusu

**Comment by akiko-pusu** [comment]
@archonwang Thanks for your confirmation and taking your time!

But I'm so sorry that I have one more bugfix (Based on Javascript).  When you select Global issue templates in a popup dialog multi time, pulldown change event is not fired. Please wait for a white.

BTW, I'm afaid this popup feature is useful or not..... Please feel free to comment :)

**Comment by archonwang** [comment]
@akiko-pusu , OK, I have suscribed it and it would informed me if you have finished.

**Comment by akiko-pusu** [comment]
@archonwang Yesterday, I’ve fixed JavaScript bug and tagged v0.1.2. Thanks.
